src/daemon.c

```diff
@@ -293,15 +291,22 @@ int become_daemon(int dont_fork, int close_all_files, const char *user, const ch
 		close(dev_null);
 
 	// generate our pid file
-	{
+	if(pidfile[0]) {
 		unlink(pidfile);
```

Comment:
ok

1. I removed the unlink() that was called at start
2. I also moved the unlink after the truncate